core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.component.ts

```diff
@@ -16,219 +16,219 @@
+import { LocalStorageService } from 'services/local-storage.service';
+import { SvgSanitizerService } from 'services/svg-sanitizer.service';
+import { TranslationLanguageService } from 'pages/exploration-editor-page/translation-tab/services/translation-language.service';
+import { TranslationTopicService } from 'pages/exploration-editor-page/translation-tab/services/translation-topic.service';
+import { UserService } from 'services/user.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+interface TabsDetails {
+  submitQuestionTab: {
+    enabled: boolean;
+  };
+}
+
+@Component({
+  selector: 'contributor-dashboard-page',
+  templateUrl: './contributor-dashboard-page.component.html'
+})
+export class ContributorDashboardPageComponent
+  implements OnInit {
+  defaultHeaderVisible: boolean;
+  profilePictureDataUrl: SafeUrl | string;
+  username: string;
+  userInfoIsLoading: boolean;
+  userIsLoggedIn: boolean;
+  userIsReviewer: boolean;
+  userCanReviewTranslationSuggestionsInLanguages: unknown[];
+  userCanReviewVoiceoverSuggestionsInLanguages: unknown[];
+  userCanReviewQuestions: boolean;
+  OPPIA_AVATAR_LINK_URL: string;
+  tabsDetails: TabsDetails;
+  OPPIA_AVATAR_IMAGE_URL: string;
+  languageCode: string;
+  topicName: string;
+  activeTabName: string;
+
+  constructor(
+    private contributionAndReviewService: ContributionAndReviewService,
+    private contributionOpportunitiesService: ContributionOpportunitiesService,
+    private focusManagerService: FocusManagerService,
+    private languageUtilService: LanguageUtilService,
+    private localStorageService: LocalStorageService,
+    private svgSanitizerService: SvgSanitizerService,
+    private translationLanguageService: TranslationLanguageService,
+    private translationTopicService: TranslationTopicService,
+    private urlInterpolationService: UrlInterpolationService,
+    private userService: UserService,
+    private windowRef: WindowRef,
+  ) {}
+
+  onTabClick(activeTabName: string): void {
+    this.activeTabName = activeTabName;
+
+    // The setTimeout is required to ensure that focus is applied only
+    // after all the functions in main thread have executed.
+    if (this.activeTabName === 'translateTextTab') {
+      setTimeout(() => {
+        this.focusManagerService.setFocusWithoutScroll('selectLangDropDown');
+      }, 5);
+    }
+  }
+
+  provideLanguageForProtractorClass(languageDescription: string): string {
+    let lang = languageDescription.split(' ').join('-').toLowerCase();
+    return lang.replace(/\(?\)?/g, '');
+  }
+
+  onChangeLanguage(languageCode: string): void {
+    this.languageCode = languageCode;
+    this.translationLanguageService.setActiveLanguageCode(this.languageCode);
+    this.localStorageService.updateLastSelectedTranslationLanguageCode(
+      this.languageCode);
+  }
+
+  showLanguageSelector(): boolean {
+    var activeTabDetail = this.tabsDetails[this.activeTabName];
```

Comment:
Prefer `const` over `var`. Here and below.

Reply:
done